src/git/zcl_abapgit_git_transport.clas.abap

```diff
@@ -6,16 +6,27 @@ CLASS zcl_abapgit_git_transport DEFINITION
   PUBLIC SECTION.
 
 * remote to local
-    CLASS-METHODS upload_pack
+    CLASS-METHODS upload_pack_by_branch
       IMPORTING
-        !iv_url         TYPE string
-        !iv_branch_name TYPE string
-        !iv_deepen      TYPE abap_bool DEFAULT abap_true
-        !it_branches    TYPE zif_abapgit_definitions=>ty_git_branch_list_tt OPTIONAL
+        !iv_url          TYPE string
+        !iv_branch_name  TYPE string
+        !iv_deepen_level TYPE numc2 DEFAULT 0
```

Comment:
earlier the default behavior was
if iv_deepen was true then the depth was 1, else everything
now the default behavior is get everything?

please roll this back to the old behavior, this change can be introduced in a follow up PR